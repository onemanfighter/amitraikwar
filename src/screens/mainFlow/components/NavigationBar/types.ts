export type NavigationBarProps = {};

export type NavigationButtonData = {
  labelKey: string;
  path: string;
};
