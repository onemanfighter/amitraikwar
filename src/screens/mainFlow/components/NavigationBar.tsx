import {
  Button,
  HStack,
  Menu,
  MenuButton,
  MenuItem,
  MenuList,
  Text,
} from '@chakra-ui/react';
import { LinkButton } from '@components';

const NavigationBar = ({
  handleCharacterClick: handlerCharacterClick,
}: {
  handleCharacterClick: (type: 'adam' | 'lieutenant') => void;
}) => {
  return (
    <HStack
      position={'fixed'}
      marginY={8}
      w={'100%'}
      px={10}
      justifyContent={'space-between'}
    >
      <LinkButton text={'amitraikwar.in'} href={'#'} fontSize={'xl'} />
      <HStack columnGap={6}>
        <LinkButton
          key={'about'}
          text={'About'}
          href={'#about'}
          fontSize={'xl'}
          animationOnHover
        />
        <LinkButton
          key={'work'}
          text={'Work'}
          href={'#work'}
          fontSize={'xl'}
          animationOnHover
        />
        <LinkButton
          key={'contact'}
          text={'Contact'}
          href={'#contact'}
          fontSize={'xl'}
          animationOnHover
        />
        <LinkButton
          key={'articles'}
          text={'Articles'}
          href={'#articles'}
          fontSize={'xl'}
          animationOnHover
        />
        <Menu>
          <MenuButton>
            <LinkButton
              key={'more'}
              text={'Character'}
              href={'#more'}
              fontSize={'xl'}
              withUnderline
            />
          </MenuButton>
          <MenuList
            style={{
              backgroundColor: 'transparent',
              color: 'violet',
              padding: '0px',
              boxShadow: '0px 0px 20px 8px violet',
            }}
          >
            <MenuItem onClick={() => handlerCharacterClick('adam')}>
              <Text key={'adam'} fontSize={'xl'}>
                Adam
              </Text>
            </MenuItem>
            <MenuItem onClick={() => handlerCharacterClick('lieutenant')}>
              <Text key={'Lieutenant'} fontSize={'xl'}>
                Lieutenant
              </Text>
            </MenuItem>
          </MenuList>
        </Menu>
      </HStack>
    </HStack>
  );
};

export default NavigationBar;
