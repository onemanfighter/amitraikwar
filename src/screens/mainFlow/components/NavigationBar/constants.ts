import { NavigationButtonData } from './types';

const NavigationButtons: NavigationButtonData[] = [
  {
    labelKey: 'MainHomeScreen.navigation.button.home',
    path: '/home',
  },
  {
    labelKey: 'MainHomeScreen.navigation.button.about',
    path: '/about',
  },
  {
    labelKey: 'MainHomeScreen.navigation.button.experience',
    path: '/experience',
  },
  {
    labelKey: 'MainHomeScreen.navigation.button.projects',
    path: '/projects',
  },
  {
    labelKey: 'MainHomeScreen.navigation.button.skills',
    path: '/skills',
  },
  {
    labelKey: 'MainHomeScreen.navigation.button.resume',
    path: '/resume',
  },
  {
    labelKey: 'MainHomeScreen.navigation.button.blogs',
    path: '/blogs',
  },
  {
    labelKey: 'MainHomeScreen.navigation.button.contact',
    path: '/contact',
  },
];

export { NavigationButtons };
