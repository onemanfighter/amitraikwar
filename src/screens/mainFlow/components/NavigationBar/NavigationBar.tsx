import { Box, Button } from '@chakra-ui/react';
import { NavigationBarProps } from './types';
import { NavigationButtons } from './constants';
import { NavLink, useLocation } from 'react-router-dom';
import { useTranslation } from 'react-i18next';
import { useColorSelector } from '@components';

const NavigationBar = (props: NavigationBarProps) => {
  const { t } = useTranslation();
  const location = useLocation();

  const { gradient } = useColorSelector();

  return (
    <Box
      display={'flex'}
      flexDirection={'row'}
      justifyContent={'center'}
      alignItems={'center'}
      padding={'4'}
    >
      <Box
        display={'flex'}
        bgGradient={gradient.topAppBar}
        padding={'6'}
        flexDirection={'row'}
        justifyContent={'center'}
        alignItems={'center'}
        width={'100%'}
        borderRadius={'lg'}
        boxShadow={'lg'}
        gap={4}
      >
        {NavigationButtons.map((button) => {
          const isCurrentPath = location.pathname === button.path;
          return (
            <Button
              key={button.labelKey}
              as={NavLink}
              to={button.path}
              padding={'6'}
              bg={isCurrentPath ? 'green.600' : 'white'}
              _hover={{
                bg: 'green.500',
                color: 'white',
                boxShadow: 'lg',
              }}
              color={isCurrentPath ? 'white' : 'black'}
              transition={'all 0.3s'}
              fontSize={'lg'}
              fontWeight={'600'}
            >
              {t(button.labelKey)}
            </Button>
          );
        })}
      </Box>
    </Box>
  );
};

export default NavigationBar;
