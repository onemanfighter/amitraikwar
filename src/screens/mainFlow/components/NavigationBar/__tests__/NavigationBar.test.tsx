import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import NavigationBar from '../NavigationBar';
import { ThemeProvider } from '@providers';

describe('NavigationBar', () => {
  it('should renders correctly', () => {
    const { container } = render(
      <ThemeProvider>
        <BrowserRouter>
          <NavigationBar />
        </BrowserRouter>
      </ThemeProvider>,
    );

    expect(container).toMatchSnapshot();
  });

  it('should show all the navigation tabs', () => {
    render(
      <ThemeProvider>
        <BrowserRouter>
          <NavigationBar />
        </BrowserRouter>
      </ThemeProvider>,
    );

    expect(screen.findByText('Home')).toBeDefined();
    expect(screen.findByText('About')).toBeDefined();
    expect(screen.findByText('Experience')).toBeDefined();
    expect(screen.findByText('Blogs')).toBeDefined();
    expect(screen.findByText('Contact')).toBeDefined();
    expect(screen.findByText('Projects')).toBeDefined();
    expect(screen.findByText('Skills')).toBeDefined();
    expect(screen.findByText('Resume')).toBeDefined();
  });
});
