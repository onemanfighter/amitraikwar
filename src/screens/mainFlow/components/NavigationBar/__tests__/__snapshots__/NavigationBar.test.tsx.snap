// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationBar should renders correctly 1`] = `
<div>
  <script
    id="chakra-script"
  >
    !(function(){try{var a=function(c){var v="(prefers-color-scheme: dark)",h=window.matchMedia(v).matches?"dark":"light",r=c==="system"?h:c,o=document.documentElement,s=document.body,l="chakra-ui-light",d="chakra-ui-dark",i=r==="dark";return s.classList.add(i?d:l),s.classList.remove(i?l:d),o.style.colorScheme=r,o.dataset.theme=r,r},n=a,m="light",e="chakra-ui-color-mode",t=localStorage.getItem(e);t?a(t):localStorage.setItem(e,a(m))}catch(a){}})();
  </script>
  <div
    class="css-zz8ks3"
  >
    <div
      class="css-y5kaw5"
    >
      <a
        class="chakra-button css-1f9j196"
        href="/home"
      >
        Home
      </a>
      <a
        class="chakra-button css-1f9j196"
        href="/about"
      >
        About
      </a>
      <a
        class="chakra-button css-1f9j196"
        href="/experience"
      >
        Experience
      </a>
      <a
        class="chakra-button css-1f9j196"
        href="/projects"
      >
        Projects
      </a>
      <a
        class="chakra-button css-1f9j196"
        href="/skills"
      >
        Skills
      </a>
      <a
        class="chakra-button css-1f9j196"
        href="/resume"
      >
        Resume
      </a>
      <a
        class="chakra-button css-1f9j196"
        href="/blogs"
      >
        Blogs
      </a>
      <a
        class="chakra-button css-1f9j196"
        href="/contact"
      >
        Contact
      </a>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
